test/compiler/codegen.jl

```diff
@@ -804,3 +804,19 @@ function f34459(args...)
     return
 end
 @test !occursin("jl_f_tuple", get_llvm(f34459, Tuple{Ptr{Int}, Type{Int}}, true, false, false))
+
+# issue #48394: incorrectly-inferred getproperty shouldn't introduce invalid cgval_t
+#               when dealing with unions of ghost values
+struct X48394
+    x::Nothing
+    y::Bool
+end
+struct Y48394
+    x::Nothing
+    z::Missing
+end
+function F48394(a, b, i)
+    c = i ? a : b
+    c.y
+end
+@test !occursin("llvm.trap", get_llvm(F48394, Tuple{X48394, Y48394, Bool}))
```

Comment:
I do in general also like calling these functions in addition to the codegen test, just to make sure the test itself is good and doesn't bitrot, but I don't want to hold up this PR further.

Reply:
Good idea, I'll add it; I messed up the test so this needs a fix anyway.